app/policies/comment_policy.rb

```diff
@@ -0,0 +1,34 @@
+class CommentPolicy < ApplicationPolicy
+  alias_method :comment, :record
+
+  def index?
+    !comment.private? || comment.resource.member?(user.profile)
+  end
+
+  def create?
+    !comment.of_comment? &&
+    (!comment.private? || comment.resource.member?(user.profile))
+  end
+
+  def update?
+    comment_owner?
+  end
+
+  def destroy?
+    comment_owner?
+  end
+
+  def permited_attributes_for_create
```

Comment:
A test is required somewhere. Preferably in policies. (This should be applied to all models using permited attributes from policies).